packages/thirdweb/src/event/actions/get-events.ts

```diff
@@ -186,3 +214,98 @@ export async function getContractEvents<
     events: resolvedEvents,
   });
 }
+
+async function getLogsFromInsight(options: {
+  params: GetLogsParamsExtra;
+  chain: Chain;
+  client: ThirdwebClient;
+  signature?: string;
+}): Promise<Log[]> {
+  const { params, chain, client, signature } = options;
+
+  const chainServices = await getChainServices(chain);
+  const insightEnabled = chainServices.some(
+    (c) => c.service === "insight" && c.enabled,
+  );
+
+  if (!insightEnabled) {
+    throw new Error(`Insight is not available for chainId ${chain.id}`);
+  }
+
+  try {
+    let baseUrl = "https://insight.thirdweb-dev.com/v1/events"; // TODO: change to prod
```

Comment:
you can use getThirdwebDomains().insight here

Reply:
done